src/aws_encryption_sdk/internal/formatting/serialize.py

```diff
@@ -227,16 +258,35 @@ def _serialize_header_auth_v2(algorithm, header, data_encryption_key, signer=Non
     :param bytes data_encryption_key: Data key with which to encrypt message
     :param signer: Cryptographic signer object (optional)
     :type signer: aws_encryption_sdk.Signer
+    :param required_encryption_context_bytes: Serialized encryption context items
+        for all items whose keys are in the required_encryption_context list.
+        This is ONLY processed if using the aws-cryptographic-materialproviders library
+        AND its required encryption context CMM. (optional)
+    :type required_encryption_context_bytes: bytes
     :returns: Serialized header authentication data
     :rtype: bytes
     """
-    header_auth = encrypt(
-        algorithm=algorithm,
-        key=data_encryption_key,
-        plaintext=b"",
-        associated_data=header,
-        iv=header_auth_iv(algorithm),
-    )
+    if required_ec_bytes is None:
+        header_auth = encrypt(
+            algorithm=algorithm,
+            key=data_encryption_key,
+            plaintext=b"",
+            associated_data=header,
+            iv=header_auth_iv(algorithm),
+        )
+    else:
+        header_auth = encrypt(
+            algorithm=algorithm,
+            key=data_encryption_key,
+            plaintext=b"",
+            # The AAD MUST be the concatenation of the serialized message header body and the serialization
+            # of encryption context to only authenticate. The encryption context to only authenticate MUST
+            # be the encryption context in the encryption materials filtered to only contain key value
+            # pairs listed in the encryption material's required encryption context keys serialized
+            # according to the encryption context serialization specification.
```

Comment:
This looks like a copy from the spec? I would suggest that you just use Duvet

Reply:
I actually copied this from ESDK-Java...
This repo doesn't have Duvet set up
I may take a look at setting it up